app/src/main/java/com/codesoom/assignment/application/TaskServiceQueries.java

```diff
@@ -0,0 +1,25 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.TaskNotFoundException;
+import com.codesoom.assignment.models.Task;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Optional;
+
+public class TaskServiceQueries {
+    private final HashMap<Long, Task> tasks;
+
+    public TaskServiceQueries(HashMap<Long, Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public Collection<Task> all() {
+        return tasks.values();
+    }
+
+    public Task details(Long id) {
+        return Optional.ofNullable(tasks.get(id))
+                .orElseThrow(() -> new TaskNotFoundException(id));
```

Comment:
안티패턴입니다. 여기에는 `Optional`이 없어도 처리가 가능하죠. 전통적인 `if`를 쓰는 것이 더 읽기 쉬우면서 더 빠른 코드가 됩니다. (`Optional.ofNullable`은 내부에서 `new` 를 하고 있습니다.) 이번 기회에 `Optional`의 javadoc을 다시 한 번 읽어보도록 합시다. `Optional`은 가능한 한 리턴 타입으로만 사용하는 것이 정석입니다. `Optional`의 javadoc에서도 설계 의도를 그렇게 밝히고 있죠.

```suggestion
        if (tasks.get(id) == null) {
            throw new TaskNotFoundException(id));
        }
        return id;
```

`Optional`을 쓰려면 다음과 같이 쓰는 것이 일반적입니다. 단, 리턴 타입을 `Task`가 아니라 `Optional<Task>`로 바꿔줘야겠죠.
```suggestion
        return Optional.ofNullable(tasks.get(id));
```